flocker_bb/steps.py

```diff
@@ -207,7 +207,15 @@ def _isMaster(branch):
         return branch == 'master'
 
     _RELEASE_TAG_RE = re.compile(
-        '^[0-9]+\.[0-9]+\.[0-9]+(?:dev[0-9]+|pre[0-9]+|\+doc[0-9]+)?$')
+        # <major>.<minor>.<micro>
+        r'^[0-9]+\.[0-9]+\.[0-9]+'
+        # weekly release
+        r'(?:dev[0-9]+'
+        # prerelease
+        r'|pre[0-9]+'
+        # documentation release
+        r'|(\+doc|\.post)[0-9]+)?'
```

Comment:
Outer parentheses are non-capturing, inner parentheses are capturing. It probably doesn't matter greatly, but we may as well be consistent.